core/templates/dev/head/pages/exploration_editor/editor_tab/StateResponses.js

```diff
@@ -595,6 +597,9 @@ oppia.controller('StateResponses', [
             $scope.trainingDataFeedback = '';
             $scope.trainingDataOutcomeDest = '';
 
+            // Inject textInputRulesService dynamically
+            textInputRulesService = $injector.get('textInputRulesService');
```

Comment:
Not quite right. Should be:

`rulesService = $injector.get(....)`

where ... is constructed by taking the interaction ID and transforming it accordingly. That keeps it general. There should be no mention of "textInput" at all in this file.